java/compaction/compaction-core/src/main/java/sleeper/compaction/strategy/CompactionStrategy.java

```diff
@@ -25,7 +26,11 @@
 
 public interface CompactionStrategy {
 
-    void init(InstanceProperties instanceProperties, TableProperties tableProperties);
+    default void init(InstanceProperties instanceProperties, TableProperties tableProperties) {
+        init(instanceProperties, tableProperties, new CompactionJobFactory(instanceProperties, tableProperties));
+    }
```

Comment:
It's already been removed in this other PR:

- https://github.com/gchq/sleeper/pull/2940

It might be easier to resolve this during the merge, so we don't have to do it twice?

Reply:
Ah ok gotcha, yeah this can be left in and i'll sort out in the merge